@xen-orchestra/lite/src/stores/xen-api.store.ts

```diff
@@ -15,7 +18,18 @@ enum STATUS {
 }
 
 export const useXenApiStore = defineStore("xen-api", () => {
-  const xenApi = new XenApi(HOST_URL);
+  const route = useRoute();
+
+  const hostUrl = computed(() => {
+    const url = new URL(HOST_URL);
+    const masterQueryUrl = getFirst(route.query["master"]);
+    if (masterQueryUrl != undefined) {
+      url.hostname = masterQueryUrl;
+    }
+    return url.origin;
+  });
+
+  const xenApi = new XenApi(hostUrl.value);
```

Comment:
Since the URL won't change on the fly, I don't think we need all these changes.

```suggestion
  const xenApi = new XenApi(useRoute().query.master ?? HOST_URL);
```